src/std_io.py

```diff
@@ -0,0 +1,167 @@
+
+    def run(self):
+        while True:
+            # read a line in hex encoding
+            line = self.inputSrc.readline()
+            if not line:
+                logger.info("STDIN: End of input")
+                if netcatExitOnEOF:
+                    shutdown.doCleanShutdown()
+                break
+
+            hexObject = line.rstrip()
+            hexTStamp = ''
+
+            # detect timestamp-tab-object format (as output by netcat mode)
+            if "\t" in hexObject:
+                hexTStamp = hexObject.split("\t")[0]
+                hexObject = hexObject.split("\t")[-1]
+
+            # unhex the input with error rejection
+            try:
+                binObject = unhexlify(hexObject)
+            except Exception:
+                logger.info("STDIN: Invalid input format")
+                continue
+
+            # sanity check on object size
+            if len(binObject) < 22:
+                logger.info("STDIN: Invalid object size")
+                continue
+
+            if not state.enableNetwork and state.enableGUI:
+                # in airgap mode, trick the status icon that we are in fact
+                #   NOT waiting for a connection
+                # (may be removed after impact analysis)
+                shared.statusIconColor = 'yellow'
+
+            if stdInputMode == 'airgap':
+                # airgap mode uses the timestamp
+                # unhex the timestamp with error rejection
+                if len(hexTStamp) == 16:
+                    try:
+                        # stdioStamp, = unpack('>Q', unhexlify(hexTStamp))
+                        _, = unpack('>Q', unhexlify(hexTStamp))
+                    except Exception:
+                        logger.info("STDIN: Invalid timestamp format: " + hexTStamp)
+                        continue
+
+            # check that proof of work is sufficient.
+            if not protocol.isProofOfWorkSufficient(binObject):
+                logger.info('STDIN: Insufficient Proof of Work')
+                continue
+
+            # extract expiry time, object type
+            eTime, = unpack('>Q', binObject[8:16])
+            objectType, = unpack('>I', binObject[16:20])
+
+            # extract version number and stream number
+            readPosition = 20  # bypass the nonce, time, and object type
+            # versionNumber, versionLength
+            _, versionLength = decodeVarint(binObject[readPosition:readPosition + 10])
+            readPosition += versionLength
+            streamNumber, streamNumberLength = decodeVarint(binObject[readPosition:readPosition + 10])
+            readPosition += streamNumberLength
+
+            # calculate inventory hash
+            inventoryHash = calculateInventoryHash(binObject)
+
+            # duplicate check on inventory hash (both netcat and airgap)
+            if inventoryHash in Inventory():
+                logger.info("STDIN: Already got object " + hexlify(inventoryHash))
```

Comment:
Not sure how many of these one might expect during 'normal' operation. Is 'info' the right log level? Maybe this and the next log statement could be debug level if it would otherwise lead to info being spammed. Maybe we expect people running in this mode to not care about other info level statements but be very interested in knowing that messages were or were not added, I don't know.

Reply:
You're probably right. It's only useful when running in manual mode and pasting objects in console, useless in most other scenarios. I'll change it to debug. 
I guess what I really need here is not `logger.info` pollution, but the ability to switch logging levels as needed via getopt; I might actually open an issue about that.